dart/lib/src/hub.dart

```diff
@@ -105,12 +113,18 @@ class Hub {
       );
     } else {
       final item = _peek();
+      var scope = item.scope;
+
+      if (withScope != null) {
+        scope = item.scope.clone();
+        withScope(scope);
+      }
```

Comment:
this probably can be a private method that returns the mutated `scope` if `withScope` is given instead of dup. the same code block across methods